Tear in Reality, Open Source Abnormality (Part 1)

Reene-Ramos opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds Tear in Reality, an Abnormality that, when breached, spawns enemies made by the community to attack the facility.

Adds Interdimensional Static, the most basic enemy TiR can spawn, and serves as a template for the open source part.

## Why It's Good For The Game

Variety, idk

## Changelog
:cl:
add: Adds Tear in Reality (_tear_in_reality internally)
add: Adds Interdimensional Static (tir_cheers internally)
add: Adds EGO Weapon and Armor (tir_claw and tir_mask internally)
/:cl:

## Did I test this?

TiR works
IS works
Weapon works
Armor does not spawn in (unknown cause)
TiR spawning IS does not work (unknown cause)
Info cabinet paper does not spawn in (unknown cause)

Reene-Ramos committed:
Adds TiR again (hopefully fixed everything)

Reene-Ramos committed:
Fixes 2 Errors, prevents potential errors, fixes TiR Spawning

Fixes the 2 errors tied to "Icon_Living"

Removed all "static" from the code to prevent potential errors

fixes a small mistake that would've made it so TiR could not spawn anything.

Adds TiR again (hopefully fixed everything)

error fix attempt

EGO Gear (needs sprites)

(causing errors) Incompleted Sprites

pre-draft edits

Apply suggestions from code review

small attack verb changed, makes TiR actually spawn stuff

fixes the EGO gear

another small fix

makes IS not a separate abnormality

(should) make it so TiR's armor actually spawns

small fixes, changes, and new armor desing

Fully sprites the EGO gear, 1 frame of the weapon

god, someone who has a better program than dream maker please help make the static effect all the sprite have.

this should FINALLY fix the weapon and armor sprites

they weren't fucking single quotes AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

renames Interdimensional Static interally

hopefully fixes errors, renames "static" to "basic static"

Finally completes Part 1

renames TiR and IS internally for file ordering

Update lobotomy-corp13.dme

Co-Authored-By: Aseidos <142242956+Aseidos@users.noreply.github.com>

Reene-Ramos committed:
error fix attempt

Reene-Ramos committed:
EGO Gear (needs sprites)

Reene-Ramos committed:
(causing errors) Incompleted Sprites

Reene-Ramos committed:
pre-draft edits

Aseidos reviewed (commented):
These changes should make the game able to compile correctly, though I have not tested the functionality of the abnormality in-game.
Apart from that, there are a few changes that are more miscellaneous and comments regarding certain aspects of the design of the abnormality and its gear.

Reene-Ramos committed:
Apply suggestions from code review

Co-authored-by: Aseidos <142242956+Aseidos@users.noreply.github.com>

Reene-Ramos committed:
small attack verb changed, makes TiR actually spawn stuff

Reene-Ramos committed:
fixes the EGO gear

Reene-Ramos committed:
another small fix

Aseidos commented on code/modules/mob/living/simple_animal/abnormality/aleph/tear_in_reality/static.dm:
```diff
@@ -0,0 +1,27 @@
+//////////////////////ONCE ENEMY IS FINISHED, PUT IN LIST AT LINE 61
+//////////////////////(picked_mob = pick())
+
+/mob/living/simple_animal/hostile/abnormality/tir_static
```
```suggestion
/mob/living/simple_animal/hostile/tir_static
```
The "abnormality" subtype is for abnormalities themselves, their minions should not have it.

Aseidos commented on code/modules/mob/living/simple_animal/abnormality/aleph/tear_in_reality/tear_in_reality.dm:
```diff
@@ -0,0 +1,111 @@
+	visible_message(span_danger("Something rises from the static..."))
+	for(var/i = 5 to 10)
+		var/turf/T = get_step(get_turf(src), pick(0, EAST))
+		var/picked_mob = /mob/living/simple_animal/hostile/abnormality/tir_static
```
```suggestion
		var/picked_mob = /mob/living/simple_animal/hostile/tir_static
```

Reene-Ramos committed:
makes IS not a separate abnormality

Reene-Ramos committed:
(should) make it so TiR's armor actually spawns

Reene-Ramos committed:
small fixes, changes, and new armor desing

Reene-Ramos reviewed (commented):
spoiler alert, it didnt.
im just gonna sprite everything else that still needs a sprite and hope it works for others.

Reene-Ramos committed:
Fully sprites the EGO gear, 1 frame of the weapon

god, someone who has a better program than dream maker please help make the static effect all the sprite have.

Reene-Ramos committed:
this should FINALLY fix the weapon and armor sprites

they weren't fucking single quotes AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

Reene-Ramos committed:
renames Interdimensional Static interally

hopefully fixes errors, renames "static" to "basic static"

Reene-Ramos committed:
Finally completes Part 1

Reene-Ramos committed:
renames TiR and IS internally for file ordering

Reene-Ramos committed:
Update lobotomy-corp13.dme

Reene-Ramos committed:
Fixes 2 Errors, prevents potential errors, fixes TiR Spawning

Fixes the 2 errors tied to "Icon_Living"

Removed all "static" from the code to prevent potential errors

fixes a small mistake that would've made it so TiR could not spawn anything.

Kitsunemitsu committed:
makes the code work

Reene-Ramos committed:
Merge branch 'Tear-in-Reality,-open-source-abno' of https://github.com/Reene-Ramos/Netzach-s-NEW-Random-Stuff into Tear-in-Reality,-open-source-abno

Reene-Ramos committed:
Redoing Kirie's changes just to make sure i didn't misread the github options

Reene-Ramos committed:
Fixes the last error

IT COMPILES

Reene-Ramos committed:
Moves the sprite to 64x96

It doesn't really need the 2 extra pixels.

Reene-Ramos committed:
A bunch of changes to (hopefully) fix shit

Reene-Ramos committed:
(fucking finally) Adds the info note

Reene-Ramos committed:
small changes & fixes wrong work type bit

Reene-Ramos committed:
Merge Conflict Fixes

Injects pieces of Star Luminary's code into this PR in order to fix a merge conflict in Aleph Melee Ego Weapons and Aleph Info

Reene-Ramos committed:
should ACTUALLY fix the Merge Conflicts

I literally just pasted TiR's code to a notepad and then replaced the entire file with the file from upstream

Reene-Ramos committed:
Flavor changes + Health Buff

Health changed; 2000 -> 5000

Changes the abnormality description

Changes the Instinct Work span_danger

Completely changes the fluff/plea note (tbh i never liked the original)

Reene-Ramos committed:
idk why this didnt get included with the previous changes

this was supposed to be in the previous changes (flavor changes + health buff)

Reene-Ramos commented:
Closing this. The idea isn't dead but thinking back on it's development, it'd probably just be better to start from scratch again.
